include/cute/algorithm/reorder.hpp

```diff
@@ -0,0 +1,296 @@
+/***************************************************************************************************
+ * Copyright (c) 2025 -----
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ **************************************************************************************************/
+#pragma once
+
+#include <cute/config.hpp>            // CUTE_HOST_DEVICE
+#include <cute/tensor_impl.hpp>       // cute::Tensor
+#include <cute/atom/reorder_atom.hpp>
+
+namespace cute
+{
+
+namespace detail {
+
+// Modify subgroup TV layout for subbyte types.
+//
+// In general on Xe successive elements in registers are assigned to work-items in
+//   round-robin order (interleaved at element granularity). However, subbyte types are
+//   only interleaved at byte granularity.
+//
+// This routine modifies the incoming layout to appear as though work-item ownership for subbyte
+//   types is also at element granularity, to uniformize later logic.
+template <class T, class InLayout>
+CUTE_HOST_DEVICE
+constexpr decltype(auto)
+subbyte_sg_tv_swizzle(const InLayout &layout)
+{
+  if constexpr (sizeof_bits_v<T> >= 8)
+    return layout;
+  else {
+    static_assert(is_static_v<InLayout>, "Layout must be static");
+    constexpr auto values = size(InLayout{}) / 16;
```

Comment:
would be easier to understand with named constant. I assume 16 here is the SG-size.